Issue #14533: Documentation fix Updated False description and Docs for AtclauseOrderCheck

MANISH-K-07 opened the pull request:
Closes #14533 

The current check description shows default order of javadoc tags as follows.

```
 * Default value is
 * {@code @author, @deprecated, @exception, @param, @return, @see, @serial, @serialData, @serialField, @since, @throws, @version}.
```
The same was reflecting in our **_outdated documentation_** : https://checkstyle.org/checks/javadoc/atclauseorder.html#Properties

After Documentation update : 

```
 * Default value is
 * {@code @author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated}.
```

MANISH-K-07 commented:
Github, generate website

MANISH-K-07 commented:
Github, generate website

MANISH-K-07 commented:
- https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=20218&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1793

- https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=20217&view=logs&j=d72e04f6-d294-5bfe-b3cd-ffcb523f40f9&t=a021e897-39dd-596d-375e-f941387b47d5&l=1249

These aren't caught on my local, the build was a success.
I must be missing an update in some file somewhere ??

mahfouz72 commented:
> I must be missing an update in some file somewhere ??

@MANISH-K-07 atclauseorder.xml.template

mahfouz72 commented:
> I must be missing an update in some file somewhere ??

@MANISH-K-07 I think atclauseorder.xml.template should be updated to

MANISH-K-07 commented:
- https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=20218&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1793

- https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=20217&view=logs&j=d72e04f6-d294-5bfe-b3cd-ffcb523f40f9&t=a021e897-39dd-596d-375e-f941387b47d5&l=1249

@romani ,
These aren't caught on my local, the build was a success.
Am I missing an update in some file somewhere ??

MANISH-K-07 commented:
- https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=20218&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1793
- https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=20218&view=logs&j=d72e04f6-d294-5bfe-b3cd-ffcb523f40f9&t=a021e897-39dd-596d-375e-f941387b47d5&l=1170

@romani ,
These errors are not showing up on my local, the build comes out a success each time I run.
Initially I thought the problem was with [atclauseorder.xml.template](https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/checks/javadoc/atclauseorder.xml.template) but the file uses macros to extract properties from AtclauseOrderCheck.java which has been already updated in class javadoc.
Am I missing an update somewhere?

MANISH-K-07 commented:
@romani , @nrmancuso,
I seem to be a bit stuck on this.. Any suggestions on what I am missing here would be helpful please

nrmancuso commented:
> @romani , @nrmancuso, I seem to be a bit stuck on this.. Any suggestions on what I am missing here would be helpful please

>These errors are not showing up on my local, the build comes out a success each time I run.

This sounds indicative of some change that is on your local, that you have not committed.

MANISH-K-07 commented:
> This sounds indicative of some change that is on your local, that you have not committed.

```
E:\MANISH\Workspace\checkstyle>git checkout atclause
Updating files: 100% (408/408), done.
Switched to branch 'atclause'

E:\MANISH\Workspace\checkstyle>git status
On branch atclause
nothing to commit, working tree clean

E:\MANISH\Workspace\checkstyle>git log --oneline
2e9ae9926 (HEAD -> atclause, origin/atclause) Issue #14533: Documentation update for atclauseorder
6d5b91f05 Issue #13345: Enabled AtClauseOrderCheckExamplesTest
```
I generally make sure to double check before pushing to remote :)
And also, the changes are reflected here but the following error/diff comes up related to gitattributes : 


```
Please clean up or update .gitattributes file.
Git status output:
Top 300 lines of diff:
HEAD detached at pull/14593/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   src/xdocs/checks/javadoc/atclauseorder.xml

no changes added to commit (use "git add" and/or "git commit -a")
diff --git a/src/xdocs/checks/javadoc/atclauseorder.xml b/src/xdocs/checks/javadoc/atclauseorder.xml
index 10ab9993d..f676aac4b 100644
--- a/src/xdocs/checks/javadoc/atclauseorder.xml
+++ b/src/xdocs/checks/javadoc/atclauseorder.xml
@@ -33,7 +33,7 @@
               <td>tagOrder</td>
               <td>Specify the order by tags.</td>
               <td><a href="../../property_types.html#String.5B.5D">String[]</a></td>
-              <td><code>@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated</code></td>
+              <td><code>@author, @deprecated, @exception, @param, @return, @see, @serial, @serialData, @serialField, @since, @throws, @version</code></td>
               <td>6.0</td>
             </tr>
             <tr>

```

nrmancuso commented:
@MANISH-K-07 I am out of ideas, keep digging in and figure it out :) I would check out a new branch to use for debugging and start with only the change in the javadoc and got from there.

romani commented:
@MANISH-K-07, do you have energy to finish this PR?

MANISH-K-07 commented:
> @MANISH-K-07, do you have energy to finish this PR?

Will start from scratch on this to let debug be easier..

romani commented:
Everyone is welcome to reuse this code and make new PR with good fix
